src/model.ts

```diff
@@ -41,31 +41,32 @@ export class HubApiModel {
     // scraping entire database
     this.validateRequestScope(searchRequest);
 
-    const pagingStreams: PagingStream[] = await getBatchedStreams(searchRequest);
+    const limit = Number(request.query?.limit) || undefined;
+    
+    const pagingStreams: PagingStream[] = await getBatchedStreams(searchRequest, limit);
 
-    let pass = new PassThrough({ objectMode: true });
-    let waiting = pagingStreams.length;
+    return this.combineStreams(pagingStreams);
```

Comment:
I would not change the current implementation if a sort is not provided. The new implementation should only be done if a sort is provided.

Reply:
@drspacemanphd I agree! separated out two ways to combine streams in sequence and not in sequence.